proto/cosmos/upgrade/v1/upgrade.proto

```diff
@@ -0,0 +1,75 @@
+syntax = "proto3";
+package cosmos.upgrade.v1;
+
+import "google/protobuf/any.proto";
+import "gogoproto/gogo.proto";
+import "google/protobuf/timestamp.proto";
+
+option go_package                      = "github.com/cosmos/cosmos-sdk/x/upgrade/types";
+option (gogoproto.goproto_getters_all) = false;
+
+// Plan specifies information about a planned upgrade and when it should occur.
+message Plan {
+  option (gogoproto.equal)            = true;
+  option (gogoproto.goproto_stringer) = false;
+
+  // Sets the name for the upgrade. This name will be used by the upgraded
+  // version of the software to apply any special "on-upgrade" commands during
+  // the first BeginBlock method after the upgrade is applied. It is also used
+  // to detect whether a software version can handle a given upgrade. If no
+  // upgrade handler with this name has been set in the software, it will be
+  // assumed that the software is out-of-date when the upgrade Time or Height is
+  // reached and the software will exit.
+  string name = 1;
+
+  // Time based upgrades have been deprecated. Time based upgrade logic
+  // has been removed from the SDK.
+  reserved 2;
+  reserved "time";
+
+  // The height at which the upgrade must be performed.
+  // Only used if Time is not set.
+  int64 height = 3;
+
+  // Any application specific upgrade info to be included on-chain
+  // such as a git commit that validators could automatically upgrade to
+  string info = 4;
+
+  // UpgradedClientState field has been deprecated. IBC upgrade logic has been
+  // moved to the IBC module in the sub module 02-client.
+  reserved 5;
+  reserved "option";
```

Comment:
Is this still needed? Since it is a new version? Any legacy IBC upgrade mechanisms will use v1beta1

Reply:
Sure, we can do that. We'll need to add a in-place store migration for x/upgrade.

Reply:
Isn't the plan deleted after (or even before) the migration?

Reply:
looks like it would happen right [after](https://github.com/cosmos/cosmos-sdk/blob/master/x/upgrade/keeper/keeper.go#L307)

I don't have strong opinions on removing/leaving the reserved field. Whatever y'all think is best. Just now seems like a time to do so